### PR TITLE
Attempt to fetch adoc version of release notes

### LINKED ISSUE #1066: Release notes procedure
In the old website, release notes are produced collectively by authors submitting PRs to this Quickbook file (or the equivalent one for the release at hand):

https://github.com/boostorg/website/blob/master/feed/history/boost_1_85_0.qbk

We need to implement a similar procedure for the new website. The ideal solution would be one where release notes are compiled automatically from individual contributions within each library repo, but for the moment being we prefer to have something conservative and minimally disrupting with respect to the protocol in place, meaning there should be a a per-version file where authors can submit their PRs.

As for the particular question of which markdown language this document should use, we can have:

* Quickbook (as now),
* Asciidoctor, or some other alternative.

I don't have a strong opinion here, if the language is anything other than Quickbook authors will probably adapt without problems.

### PR DESCRIPTION
**Pull request by brianjp93**
- if the adoc version does not exist in S3, fallback to the old process of fetching the html release notes from the boostorg repo
- fixes #1066

### CODE REVIEW

File: versions/releases.py [hunk 1]
```diff
@@ -163,7 +166,31 @@ def get_artifactory_download_data(url):
...
+            "get_release_notes_for_version_s3_error_version_not_found",
+            version_pk=version_pk,
+        )
+        raise Version.DoesNotExist
+    # get_content_from_s3 only works for keys with matching keys
+    # in the STATIC_CONTENT_MAPPING. Use get_file_data directly instead.
+    response = get_file_data(
+        get_s3_client(),
+        settings.STATIC_CONTENT_BUCKET_NAME,
+        f"release-notes/master/{version.name}.adoc",
```
Comment (brianjp93) [inline versions/releases.py:185]: I'm not sure what the exact naming convention of the files will be.  This may not be correct.

**Review by sdarwin — APPROVED** [review]
If it's working correctly...  We can probably design more tests on staging, before using the feature in production. There is also the organizational component. All boost developers will have to use the new location for release notes.     But before announcing that, experiment further on stage.boost.cppalliance.org.

**Comment by brianjp93** [comment]
I was only able to test with the existing template.adoc file in the s3 bucket, should test further when actual release notes are created.

File: versions/releases.py (hunk 1 above)
Comment (GregKaleka) [reply versions/releases.py:184]: Based on what I'm seeing in https://github.com/boostorg/website/tree/master/feed/history, it looks like the naming convention is `version.slug.replace("-", "_")`.

File: versions/releases.py [hunk 2]
```diff
@@ -163,7 +166,29 @@ def get_artifactory_download_data(url):
...
+def get_release_notes_for_version_s3(version_pk):
+    """Retrieve the adoc release notes from S3 and return the converted html string"""
+    try:
+        version = Version.objects.get(pk=version_pk)
+    except Version.DoesNotExist:
+        logger.info(
+            "get_release_notes_for_version_s3_error_version_not_found",
+            version_pk=version_pk,
+        )
+        raise Version.DoesNotExist
```
Comment (GregKaleka) [inline versions/releases.py:178]: A little cleaner, and keeps the original stack trace intact.

```suggestion
        raise
```

File: versions/releases.py [hunk 3]
```diff
@@ -195,6 +220,22 @@ def get_release_notes_for_version(version_pk):
...
+def get_release_notes_for_version(version_pk):
+    """Get the release notes content.
+
+    Tries S3 first, and fallback to old github release notes if not found in S3.
+    """
+    content = get_release_notes_for_version_s3(version_pk)
+    if content:
+        processed_content = convert_adoc_to_html(content)
+        content_type = "text/asciidoc"
+    else:
```
Comment (GregKaleka) [inline versions/releases.py:232]: Should we add a log message (either here or in the get_..._s3 function that we couldn't find the file there?

File: versions/releases.py [hunk 4]
```diff
@@ -226,17 +267,15 @@ def store_release_notes_for_version(version_pk):
         )
         raise Version.DoesNotExist
 
-    # Get the release notes content
-    content = get_release_notes_for_version(version_pk)
-    stripped_content = process_release_notes(content)
+    content, processed_content, content_type = get_release_notes_for_version(version_pk)
 
     # Save the result to the rendered content model with the version cache key
     rendered_content, _ = RenderedContent.objects.update_or_create(
```
Comment (GregKaleka) [inline versions/releases.py:273]: There doesn't appear to be any handling of the case where we fail to get the release notes - maybe I'm missing it, though?

**Review by GregKaleka — COMMENTED** [review]
Couple of small suggestions/questions.

File: versions/releases.py (hunk 3 above)
Comment (brianjp93) [reply versions/releases.py:232]: `get_file_data` logs an error message if the file is not found.

File: versions/releases.py (hunk 4 above)
Comment (brianjp93) [reply versions/releases.py:273]: There isn't, could be changed but this is identical to the previous behavior of `get_release_notes_for_version`.
This function is called in a celery task so if this function fails, just that task will fail.

Comment (GregKaleka) [reply versions/releases.py:273]: OK that's fine - we'll leave it for now.

**Review by GregKaleka — APPROVED** [review]
LGTM

**Pull request merged**